packages/protocol/contracts-0.8/common/CeloUnreleasedTreasury.sol

```diff
@@ -46,11 +58,31 @@ contract CeloUnreleasedTreasury is UsingRegistry, ReentrancyGuard, Initializable
    * @param amount The amount to release.
    */
   function release(address to, uint256 amount) external onlyEpochManager {
-    require(address(this).balance >= amount, "Insufficient balance.");
+    if (!hasAlreadyReleased) {
+      remainingBalanceToRelease = address(this).balance;
+      hasAlreadyReleased = true;
+    }
+
+    require(remainingBalanceToRelease >= amount, "Insufficient balance.");
```

Comment:
is it possible to just check remainingBalanceToRelease so we don't have to query the balance?

If not, I'd use ERC20 balance so that it can be ported to another token if needed.

Reply:
Not sure I understand what you mean by porting to another token. As is distributing a different ERC20 token instead of CELO?

Reply:
if we ever want to use this contract for another ERC20, it won't work at we use ERC20 interfaces instead of native ones.